### PR TITLE
add release-status script

### PR DESCRIPTION
**Pull request by scytacki**
The main point of this is to track down the status of PRs. The status of PT stories can be found in PT directly.

See the comments to know why each list of items is reported.

Here is an example output:
```
% npm run release-status clue-5.4.1 collaborative-learning v5.4.x master

commit date range: 2024-03-05T03:44:23Z - 2024-03-12T15:37:43Z
found 2 commits with messages that look like PR merges
found 20 PRs updated after the oldest commit
found 2 PRs with merge commits

Stories with "clue-5.4.1" 
PT-186610168: feature: Create Manual Points Legend/Toolbar Button
  PRs: 2213 State: finished
  https://www.pivotaltracker.com/story/show/186610168
PT-185581730: feature: Create Manual Points on XY Plot
  PRs: 2220 State: started
  https://www.pivotaltracker.com/story/show/185581730
PT-187179056: feature: Show as Table on Graph Tile Toolbar
  PRs:  State: unscheduled
  https://www.pivotaltracker.com/story/show/187179056

Merged PRs
2219: update CMS to Decap and React 18
  https://github.com/concord-consortium/collaborative-learning/pull/2219
  merged_at: 2024-03-12T15:37:44Z
2227: add deployment url
  https://github.com/concord-consortium/collaborative-learning/pull/2227
  merged_at: 2024-03-08T15:28:06Z

Merged PRs without a story labeled with clue-5.4.1
2219: update CMS to Decap and React 18
  https://github.com/concord-consortium/collaborative-learning/pull/2219
  merged_at: 2024-03-12T15:37:44Z
2227: add deployment url
  https://github.com/concord-consortium/collaborative-learning/pull/2227
  merged_at: 2024-03-08T15:28:06Z

Stories with "clue-5.4.1" without PRs
PT-187179056: feature: Show as Table on Graph Tile Toolbar
  PRs:  State: unscheduled
  https://www.pivotaltracker.com/story/show/187179056

Un Accepted stories with "clue-5.4.1"
PT-186610168: feature: Create Manual Points Legend/Toolbar Button
  PRs: 2213 State: finished
  https://www.pivotaltracker.com/story/show/186610168
PT-185581730: feature: Create Manual Points on XY Plot
  PRs: 2220 State: started
  https://www.pivotaltracker.com/story/show/185581730
PT-187179056: feature: Show as Table on Graph Tile Toolbar
  PRs:  State: unscheduled
  https://www.pivotaltracker.com/story/show/187179056

Open PRs linked to stories
Story: 186610168 has open PR: 2213
Story: 185581730 has open PR: 2220
```

### CODE REVIEW

File: scripts/release-status.mjs [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import 'dotenv/config'
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Octokit } from "@octokit/rest";
+
+const ptToken = process.env.PT_TOKEN;
+const ghToken = process.env.GITHUB_TOKEN;
```
Comment (dougmartin) [inline scripts/release-status.mjs:7]: Add a `console.error()` and `process.exit()` here if either token is empty?

File: scripts/release-status.mjs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+// TODO: add help text and validate command line arguments
```
Comment (dougmartin) [inline scripts/release-status.mjs:9]: Is this a todo for sometime in the future or for this PR?

File: scripts/release-status.mjs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  const msg = 
+    `PT-${story.id}: ${story.story_type}: ${name}\n` +
+    `  PRs: ${storyPrs} State: ${story.current_state}\n` +
+    `  ${story.url}`;
+  console.log(msg);
+}
+
+// use paginate with a map function so we can handle more than 250 commits
+// The result is an array of strings, even if multiple pages are returned
+// - need a git token with soemthing like `content:read`
```
Comment (dougmartin) [inline scripts/release-status.mjs:77]: s/soemthing/something/

**Review by dougmartin — APPROVED** [review]
Looks good :+1: 

A few inline questions and suggestions but nothing blocking merging this PR.

File: scripts/release-status.mjs (hunk 2 above)
Comment (scytacki) [reply scripts/release-status.mjs:9]: Something for the future. I'm not sure how useful this will be yet.

**Pull request merged**